tests/integration/server/proposals_api_test.py

```diff
@@ -13,7 +13,115 @@
+    {"name": "Teacher", "owners": "", "administrators": ""},
+]
+
+
+def setup_module():
+    """Setting up users and roles for various tests """
+    with requests.Session() as session:
+        # Creating test users
+        for user in TEST_USERS:
+            user_id = create_test_user(session, user).json()["data"]["user"]["id"]
+            user["id"] = user_id
+
+        # Creating test roles
+        for role in TEST_ROLES:
+            role["owners"] = TEST_USERS[0]["id"]
+            role["administrators"] = TEST_USERS[0]["id"]
+            role_id = create_test_role(session, role).json()["data"]["id"]
+            role["id"] = role_id
+
+
+def test_proposal_api_as_owner():
+    """Test updating a proposal as role owner"""
+    with requests.Session() as session:
+
+        # Logging in as role owner
+        credentials_payload = {
+            "id": TEST_USERS[0]["username"],
+            "password": TEST_USERS[0]["password"],
+        }
+        auth_response = log_in(session, credentials_payload)
+
+        # Creating Proposal
+        proposal_payload = {"id": TEST_USERS[1]["id"]}
+        proposal_id = add_role_member(
+            session, TEST_ROLES[0]["id"], proposal_payload
+        ).json()["proposal_id"]
+        # proposal_id = proposal_res.json()["proposal_id"]
+
+        # Updating proposal created above as role owner
+        update_proposal_payload = {
+            "status": "APPROVED",
+            "reason": "Need on team.",
+            "metadata": "approval-date:2019-05-09",
+        }
+        update_proposal_response = update_proposal(
+            session, proposal_id, update_proposal_payload
+        )
+
+        assert (
+            update_proposal_response.status_code == 200
+        ), update_proposal_response.json()
+        assert update_proposal_response.json()["proposal_id"] == proposal_id
+
+
+def test_proposal_api_as_non_owner():
+    """Test updating a proposal as non role owner"""
+    with requests.Session() as session:
+
+        # Logging in as non role owner
+        credentials_payload = {
+            "id": TEST_USERS[1]["username"],
+            "password": TEST_USERS[1]["password"],
+        }
+        auth_response = log_in(session, credentials_payload)
+
+        # Creating payload
+        proposal_payload = {"id": TEST_USERS[1]["id"]}
+        proposal_id = add_role_member(
+            session, TEST_ROLES[1]["id"], proposal_payload
+        ).json()["proposal_id"]
+        # proposal_id = proposal_res.json()["proposal_id"]
+
+        # Updating proposale created above as non role owner
+        update_proposal_payload = {
+            "status": "APPROVED",
+            "reason": "Approving myself",
+            "metadata": "approval-date:2019-05-10",
+        }
+        update_proposal_response = update_proposal(
+            session, proposal_id, update_proposal_payload
+        )
+
+        assert (
+            update_proposal_response.status_code == 400
```

Comment:
You should assert that you are receiving the correct Bad response message as well.